packages/ruleset/src/utils/validate-subschemas.js

```diff
@@ -0,0 +1,36 @@
+const validateSubschemas = (schema, path, validate) => {
```

Comment:
Can't JSONPath do all this already? Or maybe not precisely enough?

Reply:
I'm not skilled enough in JSONPath to know how to do this. The recursive strategy was already being used by two of our rules so I generalized it. Do you know how to JSONPath subschemas? Happy to alter the approach if there's a better way!

Reply:
Well, I was thinking along the lines of this (for request body schemas, as an example):

```
'$.paths[*][*][requestBody].content[*].schema',
'$.paths[*][*][requestBody].content[*].schema..properties[*]',
'$.paths[*][*][requestBody].content[*].schema..items'
```

But now that I look closely, I think that could result in false positives for object schema properties named `properties` or `items` or for the use of those key names within extensions. So... maybe there is no way around the manual recursion. 😢

On the other hand, it looks like built-in Spectral rules are willing to use schema-finding approaches that appear to have the possibility of false positives: https://github.com/stoplightio/spectral/blob/develop/packages/rulesets/src/oas/index.ts#L674